File: lab.py
import json
import requests

# Function to get the latitude and longitude of an address using MapQuest API
def get_geocode(address):
    api_key = 'kJWAsPGVWtqTOYwjFVtSrzyKD0RQpZW5'
    url = f'http://www.mapquestapi.com/geocoding/v1/address?key={api_key}&location={address}'

    try:
        response = requests.get(url)
        data = response.json()
        if data['results']:
            location = data['results'][0]['locations'][0]['latLng']
            return {'lat': location['lat'], 'lon': location['lng']}
        else:
            raise Exception('Address not found.')
    except Exception as error:
        print(error)
        raise error

# Load JSON data from file
with open('conference.json', 'r') as file:
    conference_data = json.load(file)

conferences = conference_data['conferences']

# Loop through conferences and add lat and lon
for conference in conferences:
    if 'lat' not in conference:
        address = conference['address']
        geocode = get_geocode(address)
        conference['lat'] = geocode['lat']
        conference['lon'] = geocode['lon']

# Save updated JSON data back to file
with open('conference.json', 'w') as file:
    json.dump(conference_data, file, indent=4)

print("Latitude and longitude added to conference records.")

File: lab.js
// const express = require('express');
// const fetch = require('node-fetch');
// const conferencesData = require('./conference.json');

// const app = express();
// const PORT = process.env.PORT || 3000;

// app.get('/process-conferences', async (req, res) => {
//     try {
//         // Process the conferences and update geolocation
//         for (const conference of conferencesData.conferences) {
//             try {
//                 const geocode = await getGeocode(conference.address);
//                 conference.lat = geocode.lat;
//                 conference.lon = geocode.lon;
//             } catch (error) {
//                 console.error(`Error fetching geocode for conference '${conference.name}':`, error);
//             }
//         }

//         res.status(200).json(conferencesData);
//     } catch (error) {
//         console.error(error);
//         res.status(500).json({ error: 'An error occurred' });
//     }
// });

// // Function to fetch the geocode (latitude and longitude) for an address
// async function getGeocode(address) {
//     const url = `https://nominatim.openstreetmap.org/search?format=json&q=${encodeURIComponent(address)}`;
//     const response = await fetch(url);
//     const data = await response.json();
//     if (data.length > 0) {
//         return { lat: parseFloat(data[0].lat), lon: parseFloat(data[0].lon) };
//     } else {
//         throw new Error('Address not found.');
//     }
// }

// app.listen(PORT, () => {
//     console.log(`Server is running on port ${PORT}`);
// });

await getGeocode("19 Wilson Street, Pittsburgh, PA 15223").then((data) => {
    console.log(data);
});

async function getGeocode(address) {
    const url = `https://nominatim.openstreetmap.org/search?format=json&q=${encodeURIComponent(address)}`;

    try {
        const response = await fetch(url);
        const data = await response.json();
        if (data.length > 0) {
            return { lat: parseFloat(data[0].lat), lon: parseFloat(data[0].lon) };
        } else {
            throw new Error('Address not found.');
        }
    } catch (error) {
        console.error(error);
        throw error;
    }
}

